File: test_prediction.php
<?php
require_once 'config.php';

// Test data
$test_data = [
    'age' => 30,
    'sex' => 1,
    'cp' => 1,
    'trestbps' => 160,
    'chol' => 200,
    'fbs' => 0,
    'restecg' => 0,
    'thalach' => 150,
    'exang' => 0,
    'oldpeak' => 1.0,
    'slope' => 1,
    'ca' => 0,
    'thal' => 2
];

// Create temporary file
$temp_file = TEMP_DIR . '/test_prediction_' . uniqid() . '.json';
file_put_contents($temp_file, json_encode($test_data));

// Execute prediction script
$command = sprintf('python "%s" "%s" 2>&1', PREDICT_SCRIPT, $temp_file);
$output = shell_exec($command);

// Clean up
unlink($temp_file);

// Display results
echo "Command: " . $command . "\n\n";
echo "Output:\n" . $output . "\n\n";

// Try to parse JSON
$result = json_decode(trim($output), true);
if ($result === null) {
    echo "Error: Invalid JSON output\n";
} else {
    echo "Parsed result:\n";
    print_r($result);
} 

File: test_prediction.py
import json
import sys
from pathlib import Path

# Sample data for testing
test_data = {
    "age": 22,
    "sex": 1,
    "cp": 0,
    "trestbps": 120,
    "chol": 536,
    "fbs": 1,
    "restecg": 0,
    "thalach": 90,
    "exang": 1,
    "oldpeak": 3,
    "slope": 0,
    "ca": 1,
    "thal": 0
}

# Create a temporary file with the test data
temp_file = Path("test_data.json")
with open(temp_file, "w") as f:
    json.dump(test_data, f)

# Run the prediction script
print("Running prediction with test data...")
print(f"Test data: {json.dumps(test_data, indent=2)}")

# Execute the prediction script
import subprocess
result = subprocess.run(["python", "predict.py", str(temp_file)], 
                        capture_output=True, text=True)

# Print the output
print("\nPrediction script output:")
print(result.stdout)
if result.stderr:
    print("\nErrors:")
    print(result.stderr)

# Clean up
temp_file.unlink() 